pkg/virt-controller/watch/vm.go

```diff
@@ -748,6 +775,125 @@ func (c *VMController) startVMI(vm *virtv1.VirtualMachine) error {
 	return nil
 }
 
+// Returns in seconds how long to wait before trying to start the VM again.
+func calculateStartBackoffTime(failCount int, maxDelay int) int {
+	// The algorithm is designed to work well with a dynamic maxDelay
+	// if we decide to expose this as a tuning in the future.
+	minInterval := 10
+	delaySeconds := 0
+
+	if failCount <= 0 {
+		failCount = 1
+	}
+
+	multiplier := int(math.Pow(float64(failCount), float64(2)))
+	interval := maxDelay / 30
+
+	if interval < minInterval {
+		interval = minInterval
+	}
+
+	delaySeconds = (interval * multiplier)
+	randomRange := (delaySeconds / 2) + 1
+	// add randomized seconds to offset multiple failing VMs from one another
+	delaySeconds += rand.Intn(randomRange)
+
+	if delaySeconds > maxDelay {
```

Comment:
> Otherwise we might end up with multiple failing VMs all stuck at maxDelay


we're adding jitter to the backoff which causes multiple VMs stuck in crash loops not get retried at the same time. by the time max delay is hit, all VMs which are stuck in a crash loop will at least offset from one another due to the randomization leadign up to the max delay